tools/coredump/storecoredump.go

```diff
@@ -60,6 +61,34 @@ func (scd *StoreCoredump) OpenELF(path string) (*pfelf.File, error) {
 	return scd.CoredumpProcess.OpenELF(path)
 }
 
+func (scd *StoreCoredump) ExtractAsFile(file string) (string, error) {
+	info, ok := scd.modules[file]
+	if !ok {
+		return "", fmt.Errorf("failed to open file `%s`: %w", file, os.ErrNotExist)
+	}
+
+	f, err := os.CreateTemp("", "ebpf-profiler-coredump.*")
+	if err != nil {
+		return "", fmt.Errorf("failed to extract file `%s`: %w", file, os.ErrNotExist)
```

Comment:
Does it matter that this error doesn't reflect that `CreateTemp` failed?

Reply:
Good point.
IMO, this error message should be produced by the caller of `ExtractAsFile`.

Reply:
Fixed this and the earlier/latter error as well to just return the raw error.